File: Bymed.Application/Categories/CategoryDto.cs
namespace Bymed.Application.Categories;

public sealed record CategoryDto
{
    public required Guid Id { get; init; }
    public required string Name { get; init; }
    public required string Slug { get; init; }
    public string? Description { get; init; }
    public int DisplayOrder { get; init; }
}

File: bymed-admin/src/app/shared/models/category.dto.ts
export interface CategoryDto {
  readonly id: string;
  readonly name: string;
  readonly slug: string;
  readonly description?: string;
  readonly displayOrder: number;
  readonly isActive: boolean;
}

export interface CreateCategoryRequestDto {
  readonly name: string;
  readonly slug: string;
  readonly description?: string;
  readonly displayOrder: number;
}

export interface UpdateCategoryRequestDto {
  readonly name: string;
  readonly slug: string;
  readonly description?: string;
  readonly displayOrder: number;
  readonly isActive: boolean;
}
